### DISCUSSION TITLE
Global Filtering based on row selection

### DISCUSSION THREAD

**Discussion by prlaclede** [Budibase/budibase#7419] [Help] [answered]
I have an app I would like to build using Budibase. The functionality seems very powerful but I'm not sure how to accomplish this task: 

I have one parent grid that I would like to nest several other grids under. 

Cases: 
- Case Tasks
- Case Notes
- Case Documents
- etc...

I would like to have the parent "Case" grid be selectable, and have it filter the other grids,. So if I selected "Case 1" from my first grid, the other child grids would only display the data where that foreign key relationship exists. 

Thank you for any help!

**Comment by andz-bb** [comment, marked as answer]
Hey Phillip, I have made a quick demo of what you were describing.

![Screen Recording 2022-08-23 at 16 41 37](https://user-images.githubusercontent.com/110921612/186202457-54a5c8b7-b5b2-466b-a1a8-ab6570f00231.gif)

### Data
I have three tables in this case, Cases, Tasks and Notes. Tasks and Notes are both linked to cases in a one case to many style. I imagine this is similar to how you have yours set up.

For this we need to filter the data on the Data Provider for each table that you have below the main case table. In order to be able to filter this more easily in the design stage I have made a new static formula column on the Tasks and Notes tables that contains the value of the Case _id that is linked to that row. This is the set up for that column:

![Screenshot 2022-08-23 at 16 56 20](https://user-images.githubusercontent.com/110921612/186205169-520bd9c7-e4eb-4e93-8b04-d45d348246df.png)

### Design
Here's an overview of the components in my screen:
![Screenshot 2022-08-23 at 16 58 01](https://user-images.githubusercontent.com/110921612/186205553-76a31ff4-2e61-4559-aca7-3c1a788f602e.png)

The select button on the main Cases table has an action to set the _id for the given Case row to the State. I also have a condition set up to change how the button looks when it is selected.
![Screenshot 2022-08-23 at 17 00 13](https://user-images.githubusercontent.com/110921612/186206081-996fdad4-633d-4bb1-a743-82811e200274.png)


Now that we have the Case _id easily available as a column on each of our related tables, as well as the Case ID we have selected available on our State, we can implement the filter on the Data Providers.

![Screenshot 2022-08-23 at 17 03 35](https://user-images.githubusercontent.com/110921612/186206811-b0044292-d8f5-488a-9d0e-dcf3f60f1c04.png)

Finally, my View All Button has an action set to remove caseID from the state.

Hope this made sense, if you have any questions based on anything I have said let me know and I'll be happy to help! I've included an export of my app here if you want to play around with it.

[Cases-export-1661270787897.txt](https://github.com/Budibase/budibase/files/9404304/Cases-export-1661270787897.txt)

**Reply by Kakovisk** [reply]
Excelente!!

**Comment by prlaclede** [comment]
Wow this is awesome!

You rock, thanks so much!
